### ISSUE TITLE
Too many spaces inside braces - what is the reasoning behind the rule?

### ISSUE THREAD

**Issue by dragetd** [adrienverge/yamllint#425]
There is a rule that forbids extra spaces inside braces:
https://yamllint.readthedocs.io/en/stable/rules.html#brackets

By default, `foo: [ "bar", "blupp" ]` causes a warning.

I know that most things probably come down to taste, but I wonder if there has been any convention or authority that ever said something about those spaces? The reason I am asking: The default format settings in IntelliJ IDEA are to insert spaces there!

 
![grafik](https://user-images.githubusercontent.com/3639577/142282847-3327d837-7538-4857-954d-701127b1bc3a.png)

I am just curious to know if there exist some reasoning so I can decide if I should change the default of IDEA or yamllint for that particular rule.

**Comment by dragetd** [comment]
While the spec does not define anything as being preferred, it does provide several examples that include arrays which are all written including the spaces. See https://yaml.org/spec/1.2.2/#example-flow-collection-indicators

Maybe this rule should not be part of the default configuration?

**Comment by adrienverge** [comment]
Hello,

> I know that most things probably come down to taste, but I wonder if there has been any convention or authority that ever said something about those spaces?

Not that I'm aware of. Personally, I more often read `[1, 2]` than `[ 1, 2 ]` in open-source projects I see (whatever the language).

> The reason I am asking: The default format settings in IntelliJ IDEA are to insert spaces there!

It mostly depends on the language or framework, e.g. Python vs. JavaScript. IntelliJ IDEA made a choice, but I guess it's configurable?
In yamllint, this is also configurable.

> Maybe this rule should not be part of the default configuration?

What will happen if we change the default configuration now that many many many users rely on it?

**Comment by dragetd** [comment]
Yes, I understand that. :)

Just thought someone has some insights if any authority ever said something about it. Personally, I also used to write it without, until IDEA started formatting it like this. Then yesterday, I saw the examples in the spec. I guess I'll make myself a custom yamllint config then! Thanks for the input. :-)

**Comment by kenorb** [comment]
Same error when using variables for Helm chart:

```
      volumes:
        - name: config
          configMap:
            name: {{ .Values.configMap.name }}
```

It doesn't make sense to not have extra spaces.

**Comment by andrewimeson** [comment]
@kenorb what you have posted is templated YAML. If we were to convert it to JSON, `name` would not be a simple scalar value, but a dictionary. That is why yamllint cares about it, because it sees it as part of the YAML presentation, not a template expression.

You should quote the template expressions, if Helm supports that.

```yaml
volumes:
  - name: config
    configMap:
      name: '{{ .Values.configMap.name }}'
```
